### PR TITLE
Add Blood sugar sync

### PR DESCRIPTION
**Pull request by rakshakhegde**
https://www.pivotaltracker.com/story/show/169674850

### CODE REVIEW

File: app/src/androidTest/java/org/simple/clinic/TestData.kt [hunk 1]
```diff
@@ -483,6 +485,29 @@ class TestData @Inject constructor(
...
+      updatedAt: Instant = Instant.now(),
+      deletedAt: Instant? = null,
+      recordedAt: Instant = Instant.now()
+  ) = BloodSugarMeasurementPayload(
+      uuid = uuid,
+      bloodSugarType = bloodSugarType,
+      bloodSugarValue = bloodSugarValue,
+      patientUuid = patientUuid,
+      facilityUuid = facilityUuid,
+      userUuid = qaUserUuid(),
```
Comment (vinaysshenoy) [inline app/src/androidTest/java/org/simple/clinic/TestData.kt]: Could you make this a method parameter like all the others, please?

File: app/src/androidTest/java/org/simple/clinic/bloodsugar/sync/BloodSugarSyncAndroidTest.kt [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+  @Inject
+  @field:Named("last_blood_sugar_pull_token")
+  lateinit var lastPullToken: Preference<Optional<String>>
+
+  @Inject
+  lateinit var syncApi: BloodSugarSyncApi
+
+  @Inject
+  lateinit var bpSync: BloodSugarSync
```
Comment (vinaysshenoy) [inline app/src/androidTest/java/org/simple/clinic/bloodsugar/sync/BloodSugarSyncAndroidTest.kt]: I think this was a copy-paste error, the field is still named `bpSync`.

File: app/src/androidTest/java/org/simple/clinic/TestData.kt (hunk 1 above)
Comment (rakshakhegde) [reply app/src/androidTest/java/org/simple/clinic/TestData.kt]: done.

File: app/src/main/java/org/simple/clinic/bloodsugar/sync/BloodSugarSyncApi.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+interface BloodSugarSyncApi {
+
+  @POST("blood_sugars/sync")
+  fun push(
+      @Body body: BloodSugarPushRequest
+  ): Single<DataPushResponse>
+
+  @Headers(value = ["X-RESYNC-TOKEN: 1"])
+  @GET("blood_pressures/sync")
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/bloodsugar/sync/BloodSugarSyncApi.kt]: Copy-paste error? This should be `blood_sugars/sync`.

File: app/src/androidTest/java/org/simple/clinic/bloodsugar/sync/BloodSugarSyncAndroidTest.kt (hunk 2 above)
Comment (rakshakhegde) [reply app/src/androidTest/java/org/simple/clinic/bloodsugar/sync/BloodSugarSyncAndroidTest.kt]: Done

File: app/src/main/java/org/simple/clinic/sync/SyncModule.kt [hunk 4]
```diff
@@ -80,12 +81,14 @@ class SyncModule {
       prescriptionSync: PrescriptionSync,
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/sync/SyncModule.kt]: In `SyncModule`, there is another provider, `@Named("frequently_syncing_repositories")`, which needs the blood sugar repository to be added in the list of repositories.

File: app/src/main/java/org/simple/clinic/bloodsugar/sync/BloodSugarSyncApi.kt (hunk 3 above)
Comment (rakshakhegde) [reply app/src/main/java/org/simple/clinic/bloodsugar/sync/BloodSugarSyncApi.kt]: done

File: app/src/main/java/org/simple/clinic/sync/SyncModule.kt (hunk 4 above)
Comment (rakshakhegde) [reply app/src/main/java/org/simple/clinic/sync/SyncModule.kt]: done

**Pull request merged**